tests/subsys/llext/simple/src/threads_kernel_objects_ext.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This code demonstrates the use of threads and requires object
+ * relocation support.
+ */
+
+#include <stdint.h>
+#include <zephyr/llext/symbol.h>
+#include <zephyr/kernel.h>
+
+#ifdef CONFIG_USERSPACE
+extern struct k_sem my_sem_userspace;
+extern struct k_thread my_thread_userspace;
```

Comment:
I would (re)use the local names even in the userspace case. With a few extra defines such as `MY_THREAD_PRIO`, `MY_THREAD_FLAGS` etc, most if not all of the other `#ifdefs` could go away :slightly_smiling_face: 
These `extern`s should also probably join other things in the shared .h for type safety.

Reply:
Great suggestion, it's pretty messy as-is. Have put them in a header.